---
title: "Build cellranger reference"
namespace: "Reference"
description: "Build a Cell Ranger-compatible reference folder from user-supplied genome FASTA and gene GTF files."
type: "module"
toc: false
---

::: {.column-margin}

### Info

ID: `build_cellranger_reference`  
Namespace: `reference`

### Links
[Source](https://github.com/openpipelines-bio/openpipeline/blob/1.0.1/src/reference/build_cellranger_reference/config.vsh.yaml){.btn-action .btn-sm .btn-info .btn role="button"}

:::

Creates a new folder named after the genome.

## Example commands

You can run the pipeline using `nextflow run`.

### View help

You can use `--help` as a parameter to get an overview of the possible parameters.

```bash
nextflow run openpipelines-bio/openpipeline \
  -r 1.0.1 -latest \
  -main-script target/nextflow/reference/build_cellranger_reference/main.nf \
  --help
```

### Run command

<details>
<summary>Example of `params.yaml`</summary>
```yaml
# Arguments
genome_fasta: # please fill in - example: "genome_sequence.fa.gz"
transcriptome_gtf: # please fill in - example: "transcriptome_annotation.gtf.gz"
# reference_version: "foo"
# output: "$id.$key.output.output"

# Nextflow input-output arguments
publish_dir: # please fill in - example: "output/"
# param_list: "my_params.yaml"
```
</details>

```bash
nextflow run openpipelines-bio/openpipeline \
  -r 1.0.1 -latest \
  -profile docker \
  -main-script target/nextflow/reference/build_cellranger_reference/main.nf \
  -params-file params.yaml
```

:::{.callout-note}
Replace `-profile docker` with `-profile podman` or `-profile singularity` depending on the desired backend.
:::

## Argument group

### Arguments

|Name                  |Description                                                 |Attributes                                                     |
|:----------|:--------------------------------------------------|:--------------------|
|`--genome_fasta`      |Reference genome fasta.                                     |`file`, required, example: `"genome_sequence.fa.gz"`           |
|`--transcriptome_gtf` |Reference transcriptome annotation.                         |`file`, required, example: `"transcriptome_annotation.gtf.gz"` |
|`--reference_version` |Optional reference version string to include with reference |`string`                                                       |
|`--output`            |Output folder                                               |`file`, required, example: `"cellranger_reference"`            |

## Authors


  * Angela Oliveira Pisco [{{< fa brands github >}}](https://github.com/aopisco) [{{< fa brands orcid >}}](https://orcid.org/0000-0003-0142-2355) [{{< fa brands linkedin >}}](https://linkedin.com/in/aopisco) (author)

  * Robrecht Cannoodt [{{< fa brands github >}}](https://github.com/rcannood) [{{< fa brands orcid >}}](https://orcid.org/0000-0003-3641-729X) [{{< fa brands linkedin >}}](https://linkedin.com/in/robrechtcannoodt) (author, maintainer)

