---
title: "Cellranger atac mkfastq"
namespace: "Demux"
description: "Demultiplex raw sequencing data for ATAC experiments"
type: "module"
toc: false
---

::: {.column-margin}

### Info

ID: `cellranger_atac_mkfastq`  
Namespace: `demux`

### Links
[Source](https://github.com/openpipelines-bio/openpipeline/blob/2.1.2/src/demux/cellranger_atac_mkfastq/config.vsh.yaml){.btn-action .btn-sm .btn-info .btn role="button"}

:::



## Example commands

You can run the pipeline using `nextflow run`.

### View help

You can use `--help` as a parameter to get an overview of the possible parameters.

```bash
nextflow run openpipelines-bio/openpipeline \
  -r 2.1.2 -latest \
  -main-script target/nextflow/demux/cellranger_atac_mkfastq/main.nf \
  --help
```

### Run command

<details>
<summary>Example of `params.yaml`</summary>
```yaml
# Arguments
input: # please fill in - example: "/path/to/bcl"
csv: # please fill in - example: "SampleSheet.csv"
# lanes: ["1,3"]
# use_bases_mask: ["y50n,I6n,Y50n"]
delete_undetermined: false
barcode_mismatches: 1
# output: "fastqs"
# reports: "$id.$key.reports"

# Nextflow input-output arguments
publish_dir: # please fill in - example: "output/"
# param_list: "my_params.yaml"
```
</details>

```bash
nextflow run openpipelines-bio/openpipeline \
  -r 2.1.2 -latest \
  -profile docker \
  -main-script target/nextflow/demux/cellranger_atac_mkfastq/main.nf \
  -params-file params.yaml
```

:::{.callout-note}
Replace `-profile docker` with `-profile podman` or `-profile singularity` depending on the desired backend.
:::

## Argument group

### Arguments

|Name                    |Description                                                                                                                                                                                                |Attributes                                                          |
|:----------|:--------------------------------------------------|:--------------------|
|`--input`               |Path of Illumina BCL run folder.                                                                                                                                                                           |`file`, required, example: `"/path/to/bcl"`                         |
|`--csv`                 |The path to the simple layout sample sheet.                                                                                                                                                                |`file`, required, example: `"SampleSheet.csv"`                      |
|`--lanes`               |bcl2fastq option. Semicolon-delimited series of lanes to demultiplex. Use this if you have a sample sheet for an entire flow cell but only want to generate a few lanes for further 10x Genomics analysis. |List of `string`, example: `"1,3"`, multiple_sep: `","`             |
|`--use_bases_mask`      |bcl2fastq option. Use to clip extra bases off a read if you ran extra cycles for QC.                                                                                                                       |List of `string`, example: `"y50n,I6n,Y50n"`, multiple_sep: `","`   |
|`--delete_undetermined` |bcl2fastq option. Delete the Undetermined FASTQs generated by bcl2fastq. Useful if you are demultiplexing a small number of samples from a large flow cell.                                                |`boolean_true`                                                      |
|`--barcode_mismatches`  |bcl2fastq option. Use this option to change the number of allowed mismatches per index adapter (0, 1, 2).                                                                                                  |`integer`, default: `1`                                             |
|`--output`              |The folder to store the demux results                                                                                                                                                                      |`file`, required, default: `"fastqs"`, example: `"/path/to/output"` |
|`--reports`             |Reports directory                                                                                                                                                                                          |`file`, example: `"reports_dir"`                                    |

## Authors


  * Vladimir Shitov [{{< fa brands github >}}](https://github.com/vladimirshitov) [{{< fa brands orcid >}}](https://orcid.org/0000-0002-1960-8812) [{{< fa brands linkedin >}}](https://linkedin.com/in/vladimir-shitov-9a659513b) (author)

