<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.130">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="Align fastq files using STAR.">

<title>OpenPipelines - Star align v273a</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../../../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-KKE3LFPPV8"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-KKE3LFPPV8', { 'anonymize_ip': true});
</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../images/logo.svg" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../fundamentals/index.html" rel="" target="">
 <span class="menu-text">Fundamentals</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../user_guide/index.html" rel="" target="">
 <span class="menu-text">User Guide</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../contributing/index.html" rel="" target="">
 <span class="menu-text">Contributing</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../../components/index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Reference</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../team/index.html" rel="" target="">
 <span class="menu-text">Team</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/openpipelines-bio/openpipeline" rel="" target=""><i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../components/index.html">Reference</a></li><li class="breadcrumb-item"><a href="../../../components/modules/index.html">Reference</a></li><li class="breadcrumb-item"><a href="../../../components/modules/mapping/bd_rhapsody.html">Mapping</a></li><li class="breadcrumb-item"><a href="../../../components/modules/mapping/star_align_v273a.html">Star align v273a</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../components/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reference</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../components/workflows/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflows</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Gdo</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/gdo/gdo_singlesample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Gdo singlesample</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Ingestion</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/bd_rhapsody.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BD Rhapsody</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/cellranger_mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cell Ranger mapping</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/cellranger_multi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cell Ranger multi</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/cellranger_postprocessing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cell Ranger post-processing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/conversion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Conversion</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/demux.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Demux</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/ingestion/make_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Make reference</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">Integration</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/bbknn_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bbknn leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/harmony_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Harmony leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/scanorama_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scanorama leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/scvi_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scvi leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/totalvi_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Totalvi leiden</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Common</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth4 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/common/harmony_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Harmony leiden</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
 <span class="menu-text">Initialize Integration</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth4 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/initialize_integration/initialize_integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Initialize integration</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
 <span class="menu-text">Scanorama Leiden</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth4 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/scanorama_leiden/scanorama_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scanorama leiden</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false">
 <span class="menu-text">Scvi</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth4 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/integration/scvi/scvi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scvi</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="true">
 <span class="menu-text">Multiomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/dimensionality_reduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dimensionality reduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/full_pipeline.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Full pipeline</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Integration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/multisample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multisample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/process_batches.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Process batches</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/process_samples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Process samples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/prot_multisample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prot multisample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/prot_singlesample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prot singlesample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/rna_multisample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rna multisample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/rna_singlesample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rna singlesample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/split_modalities.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Split modalities</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false">
 <span class="menu-text">Integration</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth4 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/bbknn_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bbknn leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/harmony_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Harmony leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/initialize_integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Initialize integration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/leiden_scvi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Leiden scvi</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/scanorama_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scanorama leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/scvi_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scvi leiden</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/multiomics/integration/totalvi_leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Totalvi leiden</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="true">
 <span class="menu-text">Prot</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/prot/prot_multisample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prot multisample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/prot/prot_singlesample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prot singlesample</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" aria-expanded="true">
 <span class="menu-text">Qc</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-13" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/qc/qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Qc</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-14" aria-expanded="true">
 <span class="menu-text">Rna</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-14" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-14" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/rna/rna_multisample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rna multisample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/workflows/rna/rna_singlesample.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rna singlesample</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../components/modules/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reference</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-15" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-15" class="collapse list-unstyled sidebar-section depth2 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-16" aria-expanded="true">
 <span class="menu-text">Annotate</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-16" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-16" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/annotate/popv.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Popv</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-17" aria-expanded="true">
 <span class="menu-text">Cluster</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-17" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-17" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/cluster/leiden.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Leiden</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-18" aria-expanded="true">
 <span class="menu-text">Compression</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-18" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-18" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/compression/compress_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Compress h5mu</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-19" aria-expanded="true">
 <span class="menu-text">Convert</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-19" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-19" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_10xh5_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From 10xh5 to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_10xmtx_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From 10xmtx to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_bd_to_10x_molecular_barcode_tags.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From bd to 10x molecular barcode tags</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_bdrhap_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From bdrhap to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_cellranger_multi_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From cellranger multi to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_h5ad_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From h5ad to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_h5mu_to_h5ad.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From h5mu to h5ad</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/from_h5mu_to_seurat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">From h5mu to seurat</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/convert/velocyto_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Velocyto to h5mu</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-20" aria-expanded="true">
 <span class="menu-text">Correction</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-20" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-20" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/correction/cellbender_remove_background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellbender remove background</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/correction/cellbender_remove_background_v0_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellbender remove background v0 2</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-21" aria-expanded="true">
 <span class="menu-text">Dataflow</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-21" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-21" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dataflow/concat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Concat</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dataflow/concatenate_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Concatenate h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dataflow/merge.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Merge</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dataflow/split_modalities.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Split modalities</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-22" aria-expanded="true">
 <span class="menu-text">Demux</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-22" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-22" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/demux/bcl_convert.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bcl convert</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/demux/bcl2fastq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bcl2fastq</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/demux/cellranger_mkfastq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellranger mkfastq</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-23" aria-expanded="true">
 <span class="menu-text">Dimred</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-23" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-23" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dimred/pca.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Pca</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dimred/tsne.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tsne</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/dimred/umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Umap</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-24" aria-expanded="true">
 <span class="menu-text">Download</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-24" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-24" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/download/download_file.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Download file</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/download/sync_test_resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Sync test resources</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-25" aria-expanded="true">
 <span class="menu-text">Feature Annotation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-25" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-25" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/feature_annotation/highly_variable_features_scanpy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Highly variable features scanpy</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-26" aria-expanded="true">
 <span class="menu-text">Files</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-26" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-26" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/files/make_params.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Make params</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-27" aria-expanded="true">
 <span class="menu-text">Filter</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-27" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-27" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/delimit_fraction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Delimit fraction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/do_filter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Do filter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/filter_with_counts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filter with counts</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/filter_with_hvg.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filter with hvg</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/filter_with_scrublet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filter with scrublet</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/intersect_obs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Intersect obs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/remove_modality.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Remove modality</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/filter/subset_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Subset h5mu</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-28" aria-expanded="true">
 <span class="menu-text">Genetic Demux</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-28" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-28" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/bcftools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bcftools</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/cellsnp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellsnp</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/demuxlet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Demuxlet</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/dsc_pileup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dsc pileup</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/freebayes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Freebayes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/freemuxlet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Freemuxlet</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/samtools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Samtools</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/scsplit.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scsplit</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/souporcell.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Souporcell</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/genetic_demux/vireo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Vireo</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-29" aria-expanded="true">
 <span class="menu-text">Integrate</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-29" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-29" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/integrate/harmonypy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Harmonypy</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/integrate/scanorama.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scanorama</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/integrate/scarches.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scarches</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/integrate/scvi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scvi</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/integrate/totalvi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Totalvi</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-30" aria-expanded="true">
 <span class="menu-text">Interpret</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-30" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-30" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/interpret/lianapy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lianapy</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-31" aria-expanded="true">
 <span class="menu-text">Labels Transfer</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-31" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-31" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/labels_transfer/knn.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Knn</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/labels_transfer/xgboost.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Xgboost</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-32" aria-expanded="true">
 <span class="menu-text">Mapping</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-32" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-32" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/bd_rhapsody.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BD Rhapsody</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/cellranger_count.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellranger count</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/cellranger_count_split.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellranger count split</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/cellranger_multi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellranger multi</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/htseq_count.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Htseq count</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/htseq_count_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Htseq count to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/multi_star.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multi star</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/multi_star_to_h5mu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multi star to h5mu</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/samtools_sort.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Samtools sort</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/star_align.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Star align</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/star_align_v273a.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Star align v273a</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/mapping/star_build_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Star build reference</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-33" aria-expanded="true">
 <span class="menu-text">Metadata</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-33" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-33" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/metadata/add_id.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Add id</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/metadata/grep_annotation_column.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Grep annotation column</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/metadata/join_csv.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Join csv</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/metadata/join_uns_to_obs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Join uns to obs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/metadata/move_obsm_to_obs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Move obsm to obs</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-34" aria-expanded="true">
 <span class="menu-text">Neighbors</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-34" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-34" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/neighbors/bbknn.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bbknn</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/neighbors/find_neighbors.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Find neighbors</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-35" aria-expanded="true">
 <span class="menu-text">Process 10xh5</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-35" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-35" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/process_10xh5/filter_10xh5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filter 10xh5</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-36" aria-expanded="true">
 <span class="menu-text">Qc</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-36" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-36" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/qc/calculate_qc_metrics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Calculate qc metrics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/qc/fastqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Fastqc</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/qc/multiqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multiqc</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-37" aria-expanded="true">
 <span class="menu-text">Query</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-37" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-37" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/query/cellxgene_census.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cellxgene census</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-38" aria-expanded="true">
 <span class="menu-text">Reference</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-38" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-38" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/reference/build_bdrhap_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Build bdrhap reference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/reference/build_cellranger_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Build cellranger reference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/reference/make_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Make reference</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-39" aria-expanded="true">
 <span class="menu-text">Report</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-39" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-39" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/report/mermaid.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mermaid</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-40" aria-expanded="true">
 <span class="menu-text">Transfer</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-40" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-40" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transfer/publish.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publish</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-41" aria-expanded="true">
 <span class="menu-text">Transform</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-41" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-41" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/clr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Clr</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/delete_layer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Delete layer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/log1p.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Log1p</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/move_layer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Move layer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/normalize_total.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Normalize total</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/regress_out.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Regress out</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/transform/scale.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scale</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-42" aria-expanded="true">
 <span class="menu-text">Velocity</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-42" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-42" class="collapse list-unstyled sidebar-section depth3 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/velocity/scvelo.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scvelo</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../components/modules/velocity/velocyto.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Velocyto</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Star align v273a</h1>
</div>

<div>
  <div class="description">
    Align fastq files using STAR.
  </div>
</div>


<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>


<div class="no-row-height column-margin column-container"><div class="">
<section id="info" class="level3">
<h3 class="anchored" data-anchor-id="info">Info</h3>
<p>ID: <code>star_align_v273a</code><br>
Namespace: <code>mapping</code></p>
</section>
<section id="links" class="level3">
<h3 class="anchored" data-anchor-id="links">Links</h3>
<p><a href="https://github.com/openpipelines-bio/openpipeline/blob/1.0.0/src/mapping/star_align_v273a" class="btn-action btn-sm btn-info btn" role="button">Source</a></p>
</section>
</div></div><section id="example-commands" class="level2">
<h2 class="anchored" data-anchor-id="example-commands">Example commands</h2>
<p>You can run the pipeline using <code>nextflow run</code>.</p>
<section id="view-help" class="level3">
<h3 class="anchored" data-anchor-id="view-help">View help</h3>
<p>You can use <code>--help</code> as a parameter to get an overview of the possible parameters.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run openpipelines-bio/openpipeline <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-r</span> 1.0.0 <span class="at">-latest</span> <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-main-script</span> target/nextflow/mapping/star_align_v273a/main.nf <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--help</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="run-command" class="level3">
<h3 class="anchored" data-anchor-id="run-command">Run command</h3>
<details>
<summary>
Example of <code>params.yaml</code>
</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode yaml code-with-copy"><code class="sourceCode yaml"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Input/Output</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">input</span><span class="kw">:</span><span class="co"> # please fill in - example: ["mysample_S1_L001_R1_001.fastq.gz", "mysample_S1_L001_R2_001.fastq.gz"]</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">reference</span><span class="kw">:</span><span class="co"> # please fill in - example: "/path/to/reference"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># output: "$id.$key.output.output"</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Nextflow input-output arguments</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">publish_dir</span><span class="kw">:</span><span class="co"> # please fill in - example: "output/"</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># param_list: "my_params.yaml"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run openpipelines-bio/openpipeline <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-r</span> 1.0.0 <span class="at">-latest</span> <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-profile</span> docker <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">-main-script</span> target/nextflow/mapping/star_align_v273a/main.nf <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-params-file</span> params.yaml</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Replace <code>-profile docker</code> with <code>-profile podman</code> or <code>-profile singularity</code> depending on the desired backend.</p>
</div>
</div>
</section>
</section>
<section id="argument-groups" class="level2">
<h2 class="anchored" data-anchor-id="argument-groups">Argument groups</h2>
<section id="inputoutput" class="level3">
<h3 class="anchored" data-anchor-id="inputoutput">Input/Output</h3>
<table class="table">
<colgroup>
<col style="width: 5%">
<col style="width: 42%">
<col style="width: 52%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--input</code></td>
<td style="text-align: left;">The FASTQ files to be analyzed. Corresponds to the readFilesIn in the STAR command.</td>
<td style="text-align: left;">List of <code>file</code>, required, example: <code>"mysample_S1_L001_R1_001.fastq.gz", "mysample_S1_L001_R2_001.fastq.gz"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--reference</code></td>
<td style="text-align: left;">Path to the reference built by star_build_reference. Corresponds to the genomeDir in the STAR command.</td>
<td style="text-align: left;"><code>file</code>, required, example: <code>"/path/to/reference"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--output</code></td>
<td style="text-align: left;">Path to output directory. Corresponds to the outFileNamePrefix in the STAR command.</td>
<td style="text-align: left;"><code>file</code>, required, example: <code>"/path/to/foo"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="run-parameters" class="level3">
<h3 class="anchored" data-anchor-id="run-parameters">Run Parameters</h3>
<table class="table">
<colgroup>
<col style="width: 21%">
<col style="width: 42%">
<col style="width: 36%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--runRNGseed</code></td>
<td style="text-align: left;">random number generator seed.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>777</code></td>
</tr>
</tbody>
</table>
</section>
<section id="genome-parameters" class="level3">
<h3 class="anchored" data-anchor-id="genome-parameters">Genome Parameters</h3>
<table class="table">
<colgroup>
<col style="width: 4%">
<col style="width: 84%">
<col style="width: 10%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--genomeLoad</code></td>
<td style="text-align: left;">mode of shared memory usage for the genome files. Only used with runMode alignReads. - LoadAndKeep  load genome into shared and keep it in memory after run - LoadAndRemove  load genome into shared but remove it after run - LoadAndExit  load genome into shared memory and exit, keeping the genome in memory for future runs - Remove  do not map anything, just remove loaded genome from memory - NoSharedMemory  do not use shared memory, each job will have its own private copy of the genome</td>
<td style="text-align: left;"><code>string</code>, example: <code>"NoSharedMemory"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--genomeFastaFiles</code></td>
<td style="text-align: left;">path(s) to the fasta files with the genome sequences, separated by spaces. These files should be plain text FASTA files, they <em>cannot</em> be zipped. Required for the genome generation (runMode genomeGenerate). Can also be used in the mapping (runMode alignReads) to add extra (new) sequences to the genome (e.g.&nbsp;spike-ins).</td>
<td style="text-align: left;">List of <code>file</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--genomeFileSizes</code></td>
<td style="text-align: left;">genome files exact sizes in bytes. Typically, this should not be defined by the user.</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>0</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--genomeTransformOutput</code></td>
<td style="text-align: left;">which output to transform back to original genome - SAM  SAM/BAM alignments - SJ  splice junctions (SJ.out.tab) - None  no transformation of the output</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--genomeChrSetMitochondrial</code></td>
<td style="text-align: left;">names of the mitochondrial chromosomes. Presently only used for STARsolo statistics output/</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"chrM", "M", "MT"</code>, multiple_sep: <code>";"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="splice-junctions-database" class="level3">
<h3 class="anchored" data-anchor-id="splice-junctions-database">Splice Junctions Database</h3>
<table class="table">
<colgroup>
<col style="width: 9%">
<col style="width: 68%">
<col style="width: 21%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--sjdbFileChrStartEnd</code></td>
<td style="text-align: left;">path to the files with genomic coordinates (chr <tab> start <tab> end <tab> strand) for the splice junction introns. Multiple files can be supplied and will be concatenated.</tab></tab></tab></td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--sjdbGTFfile</code></td>
<td style="text-align: left;">path to the GTF file with annotations</td>
<td style="text-align: left;"><code>file</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--sjdbGTFchrPrefix</code></td>
<td style="text-align: left;">prefix for chromosome names in a GTF file (e.g.&nbsp;chr for using ENSMEBL annotations with UCSC genomes)</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--sjdbGTFfeatureExon</code></td>
<td style="text-align: left;">feature type in GTF file to be used as exons for building transcripts</td>
<td style="text-align: left;"><code>string</code>, example: <code>"exon"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--sjdbGTFtagExonParentTranscript</code></td>
<td style="text-align: left;">GTF attribute name for parent transcript ID (default transcript_id works for GTF files)</td>
<td style="text-align: left;"><code>string</code>, example: <code>"transcript_id"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--sjdbGTFtagExonParentGene</code></td>
<td style="text-align: left;">GTF attribute name for parent gene ID (default gene_id works for GTF files)</td>
<td style="text-align: left;"><code>string</code>, example: <code>"gene_id"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--sjdbGTFtagExonParentGeneName</code></td>
<td style="text-align: left;">GTF attribute name for parent gene name</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"gene_name"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--sjdbGTFtagExonParentGeneType</code></td>
<td style="text-align: left;">GTF attribute name for parent gene type</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"gene_type", "gene_biotype"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--sjdbOverhang</code></td>
<td style="text-align: left;">length of the donor/acceptor sequence on each side of the junctions, ideally = (mate_length - 1)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>100</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--sjdbScore</code></td>
<td style="text-align: left;">extra alignment score for alignments that cross database junctions</td>
<td style="text-align: left;"><code>integer</code>, example: <code>2</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--sjdbInsertSave</code></td>
<td style="text-align: left;">which files to save when sjdb junctions are inserted on the fly at the mapping step - Basic  only small junction / transcript files - All  all files including big Genome, SA and SAindex - this will create a complete genome directory</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Basic"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="variation-parameters" class="level3">
<h3 class="anchored" data-anchor-id="variation-parameters">Variation parameters</h3>
<table class="table">
<colgroup>
<col style="width: 10%">
<col style="width: 81%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--varVCFfile</code></td>
<td style="text-align: left;">path to the VCF file that contains variation data. The 10th column should contain the genotype information, e.g.&nbsp;0/1</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
</tbody>
</table>
</section>
<section id="read-parameters" class="level3">
<h3 class="anchored" data-anchor-id="read-parameters">Read Parameters</h3>
<table class="table">
<colgroup>
<col style="width: 3%">
<col style="width: 87%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--readFilesType</code></td>
<td style="text-align: left;">format of input read files - Fastx  FASTA or FASTQ - SAM SE  SAM or BAM single-end reads; for BAM use readFilesCommand samtools view - SAM PE  SAM or BAM paired-end reads; for BAM use readFilesCommand samtools view</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Fastx"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--readFilesSAMattrKeep</code></td>
<td style="text-align: left;">for readFilesType SAM SE/PE, which SAM tags to keep in the output BAM, e.g.: readFilesSAMtagsKeep RG PL - All  keep all tags - None  do not keep any tags</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"All"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--readFilesManifest</code></td>
<td style="text-align: left;">path to the manifest file with the names of read files. The manifest file should contain 3 tab-separated columns: paired-end reads: read1_file_name <span class="math inline">\(tab\)</span> read2_file_name <span class="math inline">\(tab\)</span> read_group_line. single-end reads: read1_file_name <span class="math inline">\(tab\)</span> - <span class="math inline">\(tab\)</span> read_group_line. Spaces, but not tabs are allowed in file names. If read_group_line does not start with ID:, it can only contain one ID field, and ID: will be added to it. If read_group_line starts with ID:, it can contain several fields separated by <span class="math inline">\(tab\)</span>, and all fields will be be copied verbatim into SAM <span class="citation" data-cites="RG">@RG</span> header line.</td>
<td style="text-align: left;"><code>file</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--readFilesPrefix</code></td>
<td style="text-align: left;">prefix for the read files names, i.e.&nbsp;it will be added in front of the strings in readFilesIn</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--readFilesCommand</code></td>
<td style="text-align: left;">command line to execute for each of the input file. This command should generate FASTA or FASTQ text and send it to stdout For example: zcat - to uncompress .gz files, bzcat - to uncompress .bz2 files, etc.</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--readMapNumber</code></td>
<td style="text-align: left;">number of reads to map from the beginning of the file -1: map all reads</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--readMatesLengthsIn</code></td>
<td style="text-align: left;">Equal/NotEqual - lengths of names,sequences,qualities for both mates are the same / not the same. NotEqual is safe in all situations.</td>
<td style="text-align: left;"><code>string</code>, example: <code>"NotEqual"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--readNameSeparator</code></td>
<td style="text-align: left;">character(s) separating the part of the read names that will be trimmed in output (read name after space is always trimmed)</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"/"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--readQualityScoreBase</code></td>
<td style="text-align: left;">number to be subtracted from the ASCII code to get Phred quality score</td>
<td style="text-align: left;"><code>integer</code>, example: <code>33</code></td>
</tr>
</tbody>
</table>
</section>
<section id="read-clipping" class="level3">
<h3 class="anchored" data-anchor-id="read-clipping">Read Clipping</h3>
<table class="table">
<colgroup>
<col style="width: 6%">
<col style="width: 81%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--clipAdapterType</code></td>
<td style="text-align: left;">adapter clipping type - Hamming  adapter clipping based on Hamming distance, with the number of mismatches controlled by clip5pAdapterMMp - CellRanger4  5p and 3p adapter clipping similar to CellRanger4. Utilizes Opal package by Martin Sosic: https://github.com/Martinsos/opal - None  no adapter clipping, all other clip* parameters are disregarded</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Hamming"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--clip3pNbases</code></td>
<td style="text-align: left;">number(s) of bases to clip from 3p of each mate. If one value is given, it will be assumed the same for both mates.</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>0</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--clip3pAdapterSeq</code></td>
<td style="text-align: left;">adapter sequences to clip from 3p of each mate. If one value is given, it will be assumed the same for both mates. - polyA  polyA sequence with the length equal to read length</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--clip3pAdapterMMp</code></td>
<td style="text-align: left;">max proportion of mismatches for 3p adapter clipping for each mate. If one value is given, it will be assumed the same for both mates.</td>
<td style="text-align: left;">List of <code>double</code>, example: <code>0.1</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--clip3pAfterAdapterNbases</code></td>
<td style="text-align: left;">number of bases to clip from 3p of each mate after the adapter clipping. If one value is given, it will be assumed the same for both mates.</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>0</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--clip5pNbases</code></td>
<td style="text-align: left;">number(s) of bases to clip from 5p of each mate. If one value is given, it will be assumed the same for both mates.</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>0</code>, multiple_sep: <code>";"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="limits" class="level3">
<h3 class="anchored" data-anchor-id="limits">Limits</h3>
<table class="table">
<colgroup>
<col style="width: 10%">
<col style="width: 64%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--limitGenomeGenerateRAM</code></td>
<td style="text-align: left;">maximum available RAM (bytes) for genome generation</td>
<td style="text-align: left;"><code>long</code>, example: <code>NA</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--limitIObufferSize</code></td>
<td style="text-align: left;">max available buffers size (bytes) for input/output, per thread</td>
<td style="text-align: left;">List of <code>long</code>, example: <code>30000000, 50000000</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--limitOutSAMoneReadBytes</code></td>
<td style="text-align: left;">max size of the SAM record (bytes) for one read. Recommended value: &gt;(2<em>(LengthMate1+LengthMate2+100)</em>outFilterMultimapNmax</td>
<td style="text-align: left;"><code>long</code>, example: <code>100000</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--limitOutSJoneRead</code></td>
<td style="text-align: left;">max number of junctions for one read (including all multi-mappers)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1000</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--limitOutSJcollapsed</code></td>
<td style="text-align: left;">max number of collapsed junctions</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1000000</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--limitBAMsortRAM</code></td>
<td style="text-align: left;">maximum available RAM (bytes) for sorting BAM. If =0, it will be set to the genome index size. 0 value can only be used with genomeLoad NoSharedMemory option.</td>
<td style="text-align: left;"><code>long</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--limitSjdbInsertNsj</code></td>
<td style="text-align: left;">maximum number of junctions to be inserted to the genome on the fly at the mapping stage, including those from annotations and those detected in the 1st step of the 2-pass run</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1000000</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--limitNreadsSoft</code></td>
<td style="text-align: left;">soft limit on the number of reads</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-1</code></td>
</tr>
</tbody>
</table>
</section>
<section id="output-general" class="level3">
<h3 class="anchored" data-anchor-id="output-general">Output: general</h3>
<table class="table">
<colgroup>
<col style="width: 3%">
<col style="width: 91%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--outTmpKeep</code></td>
<td style="text-align: left;">whether to keep the temporary files after STAR runs is finished - None  remove all temporary files - All  keep all files</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outStd</code></td>
<td style="text-align: left;">which output will be directed to stdout (standard out) - Log  log messages - SAM  alignments in SAM format (which normally are output to Aligned.out.sam file), normal standard output will go into Log.std.out - BAM_Unsorted  alignments in BAM format, unsorted. Requires outSAMtype BAM Unsorted - BAM_SortedByCoordinate  alignments in BAM format, sorted by coordinate. Requires outSAMtype BAM SortedByCoordinate - BAM_Quant  alignments to transcriptome in BAM format, unsorted. Requires quantMode TranscriptomeSAM</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Log"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outReadsUnmapped</code></td>
<td style="text-align: left;">output of unmapped and partially mapped (i.e.&nbsp;mapped only one mate of a paired end read) reads in separate file(s). - None  no output - Fastx  output in separate fasta/fastq files, Unmapped.out.mate1/2</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outQSconversionAdd</code></td>
<td style="text-align: left;">add this number to the quality score (e.g.&nbsp;to convert from Illumina to Sanger, use -31)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outMultimapperOrder</code></td>
<td style="text-align: left;">order of multimapping alignments in the output files - Old_2.4  quasi-random order used before 2.5.0 - Random  random order of alignments for each multi-mapper. Read mates (pairs) are always adjacent, all alignment for each read stay together. This option will become default in the future releases.</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Old_2.4"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="output-sam-and-bam" class="level3">
<h3 class="anchored" data-anchor-id="output-sam-and-bam">Output: SAM and BAM</h3>
<table class="table">
<colgroup>
<col style="width: 1%">
<col style="width: 96%">
<col style="width: 2%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMtype</code></td>
<td style="text-align: left;">type of SAM/BAM output 1st word: - BAM  output BAM without sorting - SAM  output SAM without sorting - None  no SAM/BAM output 2nd, 3rd: - Unsorted  standard unsorted - SortedByCoordinate  sorted by coordinate. This option will allocate extra memory for sorting which can be specified by limitBAMsortRAM.</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"SAM"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMmode</code></td>
<td style="text-align: left;">mode of SAM output - None  no SAM output - Full  full SAM output - NoQS  full SAM but without quality scores</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Full"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMstrandField</code></td>
<td style="text-align: left;">Cufflinks-like strand field flag - None  not used - intronMotif  strand derived from the intron motif. This option changes the output alignments: reads with inconsistent and/or non-canonical introns are filtered out.</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMattributes</code></td>
<td style="text-align: left;">a string of desired SAM attributes, in the order desired for the output SAM. Tags can be listed in any combination/order. <strong><em>Presets: - None  no attributes - Standard  NH HI AS nM - All  NH HI AS nM NM MD jM jI MC ch </em></strong>Alignment: - NH  number of loci the reads maps to: =1 for unique mappers, &gt;1 for multimappers. Standard SAM tag. - HI  multiple alignment index, starts with outSAMattrIHstart (=1 by default). Standard SAM tag. - AS  local alignment score, +1/-1 for matches/mismateches, score* penalties for indels and gaps. For PE reads, total score for two mates. Stadnard SAM tag. - nM  number of mismatches. For PE reads, sum over two mates. - NM  edit distance to the reference (number of mismatched + inserted + deleted bases) for each mate. Standard SAM tag. - MD  string encoding mismatched and deleted reference bases (see standard SAM specifications). Standard SAM tag. - jM  intron motifs for all junctions (i.e.&nbsp;N in CIGAR): 0: non-canonical; 1: GT/AG, 2: CT/AC, 3: GC/AG, 4: CT/GC, 5: AT/AC, 6: GT/AT. If splice junctions database is used, and a junction is annotated, 20 is added to its motif value. - jI  start and end of introns for all junctions (1-based). - XS  alignment strand according to outSAMstrandField. - MC  mates CIGAR string. Standard SAM tag. - ch  marks all segment of all chimeric alingments for chimOutType WithinBAM output. - cN  number of bases clipped from the read ends: 5 and 3 <strong><em>Variation: - vA  variant allele - vG  genomic coordinate of the variant overlapped by the read. - vW  1 - alignment passes WASP filtering; 2,3,4,5,6,7 - alignment does not pass WASP filtering. Requires waspOutputMode SAMtag. </em></strong>STARsolo: - CR CY UR UY  sequences and quality scores of cell barcodes and UMIs for the solo* demultiplexing. - GX GN  gene ID and gene name for unique-gene reads. - gx gn  gene IDs and gene names for unique- and multi-gene reads. - CB UB  error-corrected cell barcodes and UMIs for solo* demultiplexing. Requires outSAMtype BAM SortedByCoordinate. - sM  assessment of CB and UMI. - sS  sequence of the entire barcode (CB,UMI,adapter). - sQ  quality of the entire barcode. ***Unsupported/undocumented: - ha  haplotype (1/2) when mapping to the diploid genome. Requires genome generated with genomeTransformType Diploid . - rB  alignment block read/genomic coordinates. - vR  read coordinate of the variant.</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"Standard"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMattrIHstart</code></td>
<td style="text-align: left;">start value for the IH attribute. 0 may be required by some downstream software, such as Cufflinks or StringTie.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMunmapped</code></td>
<td style="text-align: left;">output of unmapped reads in the SAM format 1st word: - None  no output - Within  output unmapped reads within the main SAM file (i.e.&nbsp;Aligned.out.sam) 2nd word: - KeepPairs  record unmapped mate for each alignment, and, in case of unsorted output, keep it adjacent to its mapped mate. Only affects multi-mapping reads.</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMorder</code></td>
<td style="text-align: left;">type of sorting for the SAM output Paired: one mate after the other for all paired alignments PairedKeepInputOrder: one mate after the other for all paired alignments, the order is kept the same as in the input FASTQ files</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Paired"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMprimaryFlag</code></td>
<td style="text-align: left;">which alignments are considered primary - all others will be marked with 0x100 bit in the FLAG - OneBestScore  only one alignment with the best score is primary - AllBestScore  all alignments with the best score are primary</td>
<td style="text-align: left;"><code>string</code>, example: <code>"OneBestScore"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMreadID</code></td>
<td style="text-align: left;">read ID record type - Standard  first word (until space) from the FASTx read ID line, removing /1,/2 from the end - Number  read number (index) in the FASTx file</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Standard"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMmapqUnique</code></td>
<td style="text-align: left;">0 to 255: the MAPQ value for unique mappers</td>
<td style="text-align: left;"><code>integer</code>, example: <code>255</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMflagOR</code></td>
<td style="text-align: left;">0 to 65535: sam FLAG will be bitwise ORd with this value, i.e.&nbsp;FLAG=FLAG | outSAMflagOR. This is applied after all flags have been set by STAR, and after outSAMflagAND. Can be used to set specific bits that are not set otherwise.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMflagAND</code></td>
<td style="text-align: left;">0 to 65535: sam FLAG will be bitwise ANDd with this value, i.e.&nbsp;FLAG=FLAG &amp; outSAMflagOR. This is applied after all flags have been set by STAR, but before outSAMflagOR. Can be used to unset specific bits that are not set otherwise.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>65535</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMattrRGline</code></td>
<td style="text-align: left;">SAM/BAM read group line. The first word contains the read group identifier and must start with ID:, e.g.&nbsp;outSAMattrRGline ID:xxx CN:yy DS:z z z. xxx will be added as RG tag to each output alignment. Any spaces in the tag values have to be double quoted. Comma separated RG lines correspons to different (comma separated) input files in readFilesIn. Commas have to be surrounded by spaces, e.g.&nbsp;outSAMattrRGline ID:xxx , ID:zzz DS:z z , ID:yyy DS:yyyy</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMheaderHD</code></td>
<td style="text-align: left;"><span class="citation" data-cites="HD">@HD</span> (header) line of the SAM header</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMheaderPG</code></td>
<td style="text-align: left;">extra <span class="citation" data-cites="PG">@PG</span> (software) line of the SAM header (in addition to STAR)</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMheaderCommentFile</code></td>
<td style="text-align: left;">path to the file with <span class="citation" data-cites="CO">@CO</span> (comment) lines of the SAM header</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMfilter</code></td>
<td style="text-align: left;">filter the output into main SAM/BAM files - KeepOnlyAddedReferences  only keep the reads for which all alignments are to the extra reference sequences added with genomeFastaFiles at the mapping stage. - KeepAllAddedReferences  keep all alignments to the extra reference sequences added with genomeFastaFiles at the mapping stage.</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSAMmultNmax</code></td>
<td style="text-align: left;">max number of multiple alignments for a read that will be output to the SAM/BAM files. Note that if this value is not equal to -1, the top scoring alignment will be output first - -1  all alignments (up to outFilterMultimapNmax) will be output</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSAMtlen</code></td>
<td style="text-align: left;">calculation method for the TLEN field in the SAM/BAM files - 1  leftmost base of the (+)strand mate to rightmost base of the (-)mate. (+)sign for the (+)strand mate - 2  leftmost base of any mate to rightmost base of any mate. (+)sign for the mate with the leftmost base. This is different from 1 for overlapping mates with protruding ends</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outBAMcompression</code></td>
<td style="text-align: left;">-1 to 10 BAM compression level, -1=default compression (6?), 0=no compression, 10=maximum compression</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outBAMsortingThreadN</code></td>
<td style="text-align: left;">&gt;=0: number of threads for BAM sorting. 0 will default to min(6,runThreadN).</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outBAMsortingBinsN</code></td>
<td style="text-align: left;">&gt;0: number of genome bins for coordinate-sorting</td>
<td style="text-align: left;"><code>integer</code>, example: <code>50</code></td>
</tr>
</tbody>
</table>
</section>
<section id="bam-processing" class="level3">
<h3 class="anchored" data-anchor-id="bam-processing">BAM processing</h3>
<table class="table">
<colgroup>
<col style="width: 7%">
<col style="width: 87%">
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--bamRemoveDuplicatesType</code></td>
<td style="text-align: left;">mark duplicates in the BAM file, for now only works with (i) sorted BAM fed with inputBAMfile, and (ii) for paired-end alignments only - -  no duplicate removal/marking - UniqueIdentical  mark all multimappers, and duplicate unique mappers. The coordinates, FLAG, CIGAR must be identical - UniqueIdenticalNotMulti  mark duplicate unique mappers but not multimappers.</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--bamRemoveDuplicatesMate2basesN</code></td>
<td style="text-align: left;">number of bases from the 5 of mate 2 to use in collapsing (e.g.&nbsp;for RAMPAGE)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
</tbody>
</table>
</section>
<section id="output-wiggle" class="level3">
<h3 class="anchored" data-anchor-id="output-wiggle">Output Wiggle</h3>
<table class="table">
<colgroup>
<col style="width: 6%">
<col style="width: 84%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--outWigType</code></td>
<td style="text-align: left;">type of signal output, e.g.&nbsp;bedGraph OR bedGraph read1_5p. Requires sorted BAM: outSAMtype BAM SortedByCoordinate . 1st word: - None  no signal output - bedGraph  bedGraph format - wiggle  wiggle format 2nd word: - read1_5p  signal from only 5 of the 1st read, useful for CAGE/RAMPAGE etc - read2  signal from only 2nd read</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outWigStrand</code></td>
<td style="text-align: left;">strandedness of wiggle/bedGraph output - Stranded  separate strands, str1 and str2 - Unstranded  collapsed strands</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Stranded"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outWigReferencesPrefix</code></td>
<td style="text-align: left;">prefix matching reference names to include in the output wiggle file, e.g.&nbsp;chr, default - - include all references</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outWigNorm</code></td>
<td style="text-align: left;">type of normalization for the signal - RPM  reads per million of mapped reads - None  no normalization, raw counts</td>
<td style="text-align: left;"><code>string</code>, example: <code>"RPM"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="output-filtering" class="level3">
<h3 class="anchored" data-anchor-id="output-filtering">Output Filtering</h3>
<table class="table">
<colgroup>
<col style="width: 7%">
<col style="width: 82%">
<col style="width: 10%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--outFilterType</code></td>
<td style="text-align: left;">type of filtering - Normal  standard filtering using only current alignment - BySJout  keep only those reads that contain junctions that passed filtering into SJ.out.tab</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Normal"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outFilterMultimapScoreRange</code></td>
<td style="text-align: left;">the score range below the maximum score for multimapping alignments</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outFilterMultimapNmax</code></td>
<td style="text-align: left;">maximum number of loci the read is allowed to map to. Alignments (all of them) will be output only if the read maps to no more loci than this value. Otherwise no alignments will be output, and the read will be counted as mapped to too many loci in the Log.final.out .</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outFilterMismatchNmax</code></td>
<td style="text-align: left;">alignment will be output only if it has no more mismatches than this value.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outFilterMismatchNoverLmax</code></td>
<td style="text-align: left;">alignment will be output only if its ratio of mismatches to <em>mapped</em> length is less than or equal to this value.</td>
<td style="text-align: left;"><code>double</code>, example: <code>0.3</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outFilterMismatchNoverReadLmax</code></td>
<td style="text-align: left;">alignment will be output only if its ratio of mismatches to <em>read</em> length is less than or equal to this value.</td>
<td style="text-align: left;"><code>double</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outFilterScoreMin</code></td>
<td style="text-align: left;">alignment will be output only if its score is higher than or equal to this value.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outFilterScoreMinOverLread</code></td>
<td style="text-align: left;">same as outFilterScoreMin, but normalized to read length (sum of mates lengths for paired-end reads)</td>
<td style="text-align: left;"><code>double</code>, example: <code>0.66</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outFilterMatchNmin</code></td>
<td style="text-align: left;">alignment will be output only if the number of matched bases is higher than or equal to this value.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outFilterMatchNminOverLread</code></td>
<td style="text-align: left;">sam as outFilterMatchNmin, but normalized to the read length (sum of mates lengths for paired-end reads).</td>
<td style="text-align: left;"><code>double</code>, example: <code>0.66</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outFilterIntronMotifs</code></td>
<td style="text-align: left;">filter alignment using their motifs - None  no filtering - RemoveNoncanonical  filter out alignments that contain non-canonical junctions - RemoveNoncanonicalUnannotated  filter out alignments that contain non-canonical unannotated junctions when using annotated splice junctions database. The annotated non-canonical junctions will be kept.</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outFilterIntronStrands</code></td>
<td style="text-align: left;">filter alignments - RemoveInconsistentStrands  remove alignments that have junctions with inconsistent strands - None  no filtering</td>
<td style="text-align: left;"><code>string</code>, example: <code>"RemoveInconsistentStrands"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="output-splice-junctions-sj.out.tab" class="level3">
<h3 class="anchored" data-anchor-id="output-splice-junctions-sj.out.tab">Output splice junctions (SJ.out.tab)</h3>
<table class="table">
<colgroup>
<col style="width: 8%">
<col style="width: 72%">
<col style="width: 19%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--outSJtype</code></td>
<td style="text-align: left;">type of splice junction output - Standard  standard SJ.out.tab output - None  no splice junction output</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Standard"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="output-filtering-splice-junctions" class="level3">
<h3 class="anchored" data-anchor-id="output-filtering-splice-junctions">Output Filtering: Splice Junctions</h3>
<table class="table">
<colgroup>
<col style="width: 6%">
<col style="width: 77%">
<col style="width: 15%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--outSJfilterReads</code></td>
<td style="text-align: left;">which reads to consider for collapsed splice junctions output - All  all reads, unique- and multi-mappers - Unique  uniquely mapping reads only</td>
<td style="text-align: left;"><code>string</code>, example: <code>"All"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSJfilterOverhangMin</code></td>
<td style="text-align: left;">minimum overhang length for splice junctions on both sides for: (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif. -1 means no output for that motif does not apply to annotated junctions</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>30, 12, 12, 12</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSJfilterCountUniqueMin</code></td>
<td style="text-align: left;">minimum uniquely mapping read count per junction for: (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif. -1 means no output for that motif Junctions are output if one of outSJfilterCountUniqueMin OR outSJfilterCountTotalMin conditions are satisfied does not apply to annotated junctions</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>3, 1, 1, 1</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSJfilterCountTotalMin</code></td>
<td style="text-align: left;">minimum total (multi-mapping+unique) read count per junction for: (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif. -1 means no output for that motif Junctions are output if one of outSJfilterCountUniqueMin OR outSJfilterCountTotalMin conditions are satisfied does not apply to annotated junctions</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>3, 1, 1, 1</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--outSJfilterDistToOtherSJmin</code></td>
<td style="text-align: left;">minimum allowed distance to other junctions donor/acceptor does not apply to annotated junctions</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>10, 0, 5, 10</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--outSJfilterIntronMaxVsReadN</code></td>
<td style="text-align: left;">maximum gap allowed for junctions supported by 1,2,3,,,N reads i.e.&nbsp;by default junctions supported by 1 read can have gaps &lt;=50000b, by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. by &gt;=4 reads any gap &lt;=alignIntronMax does not apply to annotated junctions</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>50000, 100000, 200000</code>, multiple_sep: <code>";"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="scoring" class="level3">
<h3 class="anchored" data-anchor-id="scoring">Scoring</h3>
<table class="table">
<colgroup>
<col style="width: 17%">
<col style="width: 67%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--scoreGap</code></td>
<td style="text-align: left;">splice junction penalty (independent on intron motif)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--scoreGapNoncan</code></td>
<td style="text-align: left;">non-canonical junction penalty (in addition to scoreGap)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-8</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--scoreGapGCAG</code></td>
<td style="text-align: left;">GC/AG and CT/GC junction penalty (in addition to scoreGap)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-4</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--scoreGapATAC</code></td>
<td style="text-align: left;">AT/AC and GT/AT junction penalty (in addition to scoreGap)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-8</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--scoreGenomicLengthLog2scale</code></td>
<td style="text-align: left;">extra score logarithmically scaled with genomic length of the alignment: scoreGenomicLengthLog2scale*log2(genomicLength)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--scoreDelOpen</code></td>
<td style="text-align: left;">deletion open penalty</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-2</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--scoreDelBase</code></td>
<td style="text-align: left;">deletion extension penalty per base (in addition to scoreDelOpen)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-2</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--scoreInsOpen</code></td>
<td style="text-align: left;">insertion open penalty</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-2</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--scoreInsBase</code></td>
<td style="text-align: left;">insertion extension penalty per base (in addition to scoreInsOpen)</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-2</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--scoreStitchSJshift</code></td>
<td style="text-align: left;">maximum score reduction while searching for SJ boundaries in the stitching step</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
</tbody>
</table>
</section>
<section id="alignments-and-seeding" class="level3">
<h3 class="anchored" data-anchor-id="alignments-and-seeding">Alignments and Seeding</h3>
<table class="table">
<colgroup>
<col style="width: 7%">
<col style="width: 80%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--seedSearchStartLmax</code></td>
<td style="text-align: left;">defines the search start point through the read - the read is split into pieces no longer than this value</td>
<td style="text-align: left;"><code>integer</code>, example: <code>50</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--seedSearchStartLmaxOverLread</code></td>
<td style="text-align: left;">seedSearchStartLmax normalized to read length (sum of mates lengths for paired-end reads)</td>
<td style="text-align: left;"><code>double</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--seedSearchLmax</code></td>
<td style="text-align: left;">defines the maximum length of the seeds, if =0 seed length is not limited</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--seedMultimapNmax</code></td>
<td style="text-align: left;">only pieces that map fewer than this value are utilized in the stitching procedure</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10000</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--seedPerReadNmax</code></td>
<td style="text-align: left;">max number of seeds per read</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1000</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--seedPerWindowNmax</code></td>
<td style="text-align: left;">max number of seeds per window</td>
<td style="text-align: left;"><code>integer</code>, example: <code>50</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--seedNoneLociPerWindow</code></td>
<td style="text-align: left;">max number of one seed loci per window</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--seedSplitMin</code></td>
<td style="text-align: left;">min length of the seed sequences split by Ns or mate gap</td>
<td style="text-align: left;"><code>integer</code>, example: <code>12</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--seedMapMin</code></td>
<td style="text-align: left;">min length of seeds to be mapped</td>
<td style="text-align: left;"><code>integer</code>, example: <code>5</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignIntronMin</code></td>
<td style="text-align: left;">minimum intron size, genomic gap is considered intron if its length&gt;=alignIntronMin, otherwise it is considered Deletion</td>
<td style="text-align: left;"><code>integer</code>, example: <code>21</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignIntronMax</code></td>
<td style="text-align: left;">maximum intron size, if 0, max intron size will be determined by (2^winBinNbits)*winAnchorDistNbins</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignMatesGapMax</code></td>
<td style="text-align: left;">maximum gap between two mates, if 0, max intron gap will be determined by (2^winBinNbits)*winAnchorDistNbins</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignSJoverhangMin</code></td>
<td style="text-align: left;">minimum overhang (i.e.&nbsp;block size) for spliced alignments</td>
<td style="text-align: left;"><code>integer</code>, example: <code>5</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignSJstitchMismatchNmax</code></td>
<td style="text-align: left;">maximum number of mismatches for stitching of the splice junctions (-1: no limit). (1) non-canonical motifs, (2) GT/AG and CT/AC motif, (3) GC/AG and CT/GC motif, (4) AT/AC and GT/AT motif.</td>
<td style="text-align: left;">List of <code>integer</code>, example: <code>0, -1, 0, 0</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignSJDBoverhangMin</code></td>
<td style="text-align: left;">minimum overhang (i.e.&nbsp;block size) for annotated (sjdb) spliced alignments</td>
<td style="text-align: left;"><code>integer</code>, example: <code>3</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignSplicedMateMapLmin</code></td>
<td style="text-align: left;">minimum mapped length for a read mate that is spliced</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignSplicedMateMapLminOverLmate</code></td>
<td style="text-align: left;">alignSplicedMateMapLmin normalized to mate length</td>
<td style="text-align: left;"><code>double</code>, example: <code>0.66</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignWindowsPerReadNmax</code></td>
<td style="text-align: left;">max number of windows per read</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10000</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignTranscriptsPerWindowNmax</code></td>
<td style="text-align: left;">max number of transcripts per window</td>
<td style="text-align: left;"><code>integer</code>, example: <code>100</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignTranscriptsPerReadNmax</code></td>
<td style="text-align: left;">max number of different alignments per read to consider</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10000</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignEndsType</code></td>
<td style="text-align: left;">type of read ends alignment - Local  standard local alignment with soft-clipping allowed - EndToEnd  force end-to-end read alignment, do not soft-clip - Extend5pOfRead1  fully extend only the 5p of the read1, all other ends: local alignment - Extend5pOfReads12  fully extend only the 5p of the both read1 and read2, all other ends: local alignment</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Local"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignEndsProtrude</code></td>
<td style="text-align: left;">allow protrusion of alignment ends, i.e.&nbsp;start (end) of the +strand mate downstream of the start (end) of the -strand mate 1st word: int: maximum number of protrusion bases allowed 2nd word: string: - ConcordantPair  report alignments with non-zero protrusion as concordant pairs - DiscordantPair  report alignments with non-zero protrusion as discordant pairs</td>
<td style="text-align: left;"><code>string</code>, example: <code>"0    ConcordantPair"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--alignSoftClipAtReferenceEnds</code></td>
<td style="text-align: left;">allow the soft-clipping of the alignments past the end of the chromosomes - Yes  allow - No  prohibit, useful for compatibility with Cufflinks</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Yes"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--alignInsertionFlush</code></td>
<td style="text-align: left;">how to flush ambiguous insertion positions - None  insertions are not flushed - Right  insertions are flushed to the right</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
</tbody>
</table>
</section>
<section id="paired-end-reads" class="level3">
<h3 class="anchored" data-anchor-id="paired-end-reads">Paired-End reads</h3>
<table class="table">
<colgroup>
<col style="width: 11%">
<col style="width: 75%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--peOverlapNbasesMin</code></td>
<td style="text-align: left;">minimum number of overlapping bases to trigger mates merging and realignment. Specify &gt;0 value to switch on the merginf of overlapping mates algorithm.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--peOverlapMMp</code></td>
<td style="text-align: left;">maximum proportion of mismatched bases in the overlap area</td>
<td style="text-align: left;"><code>double</code>, example: <code>0.01</code></td>
</tr>
</tbody>
</table>
</section>
<section id="windows-anchors-binning" class="level3">
<h3 class="anchored" data-anchor-id="windows-anchors-binning">Windows, Anchors, Binning</h3>
<table class="table">
<colgroup>
<col style="width: 16%">
<col style="width: 69%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--winAnchorMultimapNmax</code></td>
<td style="text-align: left;">max number of loci anchors are allowed to map to</td>
<td style="text-align: left;"><code>integer</code>, example: <code>50</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--winBinNbits</code></td>
<td style="text-align: left;">=log2(winBin), where winBin is the size of the bin for the windows/clustering, each window will occupy an integer number of bins.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>16</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--winAnchorDistNbins</code></td>
<td style="text-align: left;">max number of bins between two anchors that allows aggregation of anchors into one window</td>
<td style="text-align: left;"><code>integer</code>, example: <code>9</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--winFlankNbins</code></td>
<td style="text-align: left;">log2(winFlank), where win Flank is the size of the left and right flanking regions for each window</td>
<td style="text-align: left;"><code>integer</code>, example: <code>4</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--winReadCoverageRelativeMin</code></td>
<td style="text-align: left;">minimum relative coverage of the read sequence by the seeds in a window, for STARlong algorithm only.</td>
<td style="text-align: left;"><code>double</code>, example: <code>0.5</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--winReadCoverageBasesMin</code></td>
<td style="text-align: left;">minimum number of bases covered by the seeds in a window , for STARlong algorithm only.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
</tbody>
</table>
</section>
<section id="chimeric-alignments" class="level3">
<h3 class="anchored" data-anchor-id="chimeric-alignments">Chimeric Alignments</h3>
<table class="table">
<colgroup>
<col style="width: 5%">
<col style="width: 82%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--chimOutType</code></td>
<td style="text-align: left;">type of chimeric output - Junctions  Chimeric.out.junction - SeparateSAMold  output old SAM into separate Chimeric.out.sam file - WithinBAM  output into main aligned BAM files (Aligned.*.bam) - WithinBAM HardClip  (default) hard-clipping in the CIGAR for supplemental chimeric alignments (default if no 2nd word is present) - WithinBAM SoftClip  soft-clipping in the CIGAR for supplemental chimeric alignments</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"Junctions"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimSegmentMin</code></td>
<td style="text-align: left;">minimum length of chimeric segment length, if ==0, no chimeric output</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--chimScoreMin</code></td>
<td style="text-align: left;">minimum total (summed) score of the chimeric segments</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimScoreDropMax</code></td>
<td style="text-align: left;">max drop (difference) of chimeric score (the sum of scores of all chimeric segments) from the read length</td>
<td style="text-align: left;"><code>integer</code>, example: <code>20</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--chimScoreSeparation</code></td>
<td style="text-align: left;">minimum difference (separation) between the best chimeric score and the next one</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimScoreJunctionNonGTAG</code></td>
<td style="text-align: left;">penalty for a non-GT/AG chimeric junction</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--chimJunctionOverhangMin</code></td>
<td style="text-align: left;">minimum overhang for a chimeric junction</td>
<td style="text-align: left;"><code>integer</code>, example: <code>20</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimSegmentReadGapMax</code></td>
<td style="text-align: left;">maximum gap in the read sequence between chimeric segments</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--chimFilter</code></td>
<td style="text-align: left;">different filters for chimeric alignments - None  no filtering - banGenomicN  Ns are not allowed in the genome sequence around the chimeric junction</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"banGenomicN"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimMainSegmentMultNmax</code></td>
<td style="text-align: left;">maximum number of multi-alignments for the main chimeric segment. =1 will prohibit multimapping main segments.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--chimMultimapNmax</code></td>
<td style="text-align: left;">maximum number of chimeric multi-alignments - 0  use the old scheme for chimeric detection which only considered unique alignments</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimMultimapScoreRange</code></td>
<td style="text-align: left;">the score range for multi-mapping chimeras below the best chimeric score. Only works with chimMultimapNmax &gt; 1</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--chimNonchimScoreDropMin</code></td>
<td style="text-align: left;">to trigger chimeric detection, the drop in the best non-chimeric alignment score with respect to the read length has to be greater than this value</td>
<td style="text-align: left;"><code>integer</code>, example: <code>20</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--chimOutJunctionFormat</code></td>
<td style="text-align: left;">formatting type for the Chimeric.out.junction file - 0  no comment lines/headers - 1  comment lines at the end of the file: command line and Nreads: total, unique/multi-mapping</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
</tbody>
</table>
</section>
<section id="quantification-of-annotations" class="level3">
<h3 class="anchored" data-anchor-id="quantification-of-annotations">Quantification of Annotations</h3>
<table class="table">
<colgroup>
<col style="width: 12%">
<col style="width: 71%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--quantMode</code></td>
<td style="text-align: left;">types of quantification requested - -  none - TranscriptomeSAM  output SAM/BAM alignments to transcriptome into a separate file - GeneCounts  count reads per gene</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--quantTranscriptomeBAMcompression</code></td>
<td style="text-align: left;">-2 to 10 transcriptome BAM compression level - -2  no BAM output - -1  default compression (6?) - 0  no compression - 10  maximum compression</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--quantTranscriptomeBan</code></td>
<td style="text-align: left;">prohibit various alignment type - IndelSoftclipSingleend  prohibit indels, soft clipping and single-end alignments - compatible with RSEM - Singleend  prohibit single-end alignments</td>
<td style="text-align: left;"><code>string</code>, example: <code>"IndelSoftclipSingleend"</code></td>
</tr>
</tbody>
</table>
</section>
<section id="pass-mapping" class="level3">
<h3 class="anchored" data-anchor-id="pass-mapping">2-pass Mapping</h3>
<table class="table">
<colgroup>
<col style="width: 9%">
<col style="width: 79%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--twopassMode</code></td>
<td style="text-align: left;">2-pass mapping mode. - None  1-pass mapping - Basic  basic 2-pass mapping, with all 1st pass junctions inserted into the genome indices on the fly</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--twopass1readsN</code></td>
<td style="text-align: left;">number of reads to process for the 1st step. Use very large number (or default -1) to map all reads in the first step.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>-1</code></td>
</tr>
</tbody>
</table>
</section>
<section id="wasp-parameters" class="level3">
<h3 class="anchored" data-anchor-id="wasp-parameters">WASP parameters</h3>
<table class="table">
<colgroup>
<col style="width: 4%">
<col style="width: 92%">
<col style="width: 2%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--waspOutputMode</code></td>
<td style="text-align: left;">WASP allele-specific output type. This is re-implementation of the original WASP mappability filtering by Bryce van de Geijn, Graham McVicker, Yoav Gilad &amp; Jonathan K Pritchard. Please cite the original WASP paper: Nature Methods 12, 1061-1063 (2015), https://www.nature.com/articles/nmeth.3582 . - SAMtag  add WASP tags to the alignments that pass WASP filtering</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
</tbody>
</table>
</section>
<section id="starsolo-single-cell-rna-seq-parameters" class="level3">
<h3 class="anchored" data-anchor-id="starsolo-single-cell-rna-seq-parameters">STARsolo (single cell RNA-seq) parameters</h3>
<table class="table">
<colgroup>
<col style="width: 2%">
<col style="width: 88%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Attributes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>--soloType</code></td>
<td style="text-align: left;">type of single-cell RNA-seq - CB_UMI_Simple  (a.k.a. Droplet) one UMI and one Cell Barcode of fixed length in read2, e.g.&nbsp;Drop-seq and 10X Chromium. - CB_UMI_Complex  multiple Cell Barcodes of varying length, one UMI of fixed length and one adapter sequence of fixed length are allowed in read2 only (e.g.&nbsp;inDrop, ddSeq). - CB_samTagOut  output Cell Barcode as CR and/or CB SAm tag. No UMI counting. readFilesIn cDNA_read1 [cDNA_read2 if paired-end] CellBarcode_read . Requires outSAMtype BAM Unsorted [and/or SortedByCoordinate] - SmartSeq  Smart-seq: each cell in a separate FASTQ (paired- or single-end), barcodes are corresponding read-groups, no UMI sequences, alignments deduplicated according to alignment start and end (after extending soft-clipped bases)</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloCBwhitelist</code></td>
<td style="text-align: left;">file(s) with whitelist(s) of cell barcodes. Only soloType CB_UMI_Complex allows more than one whitelist file. - None  no whitelist: all cell barcodes are allowed</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloCBstart</code></td>
<td style="text-align: left;">cell barcode start base</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloCBlen</code></td>
<td style="text-align: left;">cell barcode length</td>
<td style="text-align: left;"><code>integer</code>, example: <code>16</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloUMIstart</code></td>
<td style="text-align: left;">UMI start base</td>
<td style="text-align: left;"><code>integer</code>, example: <code>17</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloUMIlen</code></td>
<td style="text-align: left;">UMI length</td>
<td style="text-align: left;"><code>integer</code>, example: <code>10</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloBarcodeReadLength</code></td>
<td style="text-align: left;">length of the barcode read - 1  equal to sum of soloCBlen+soloUMIlen - 0  not defined, do not check</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloBarcodeMate</code></td>
<td style="text-align: left;">identifies which read mate contains the barcode (CB+UMI) sequence - 0  barcode sequence is on separate read, which should always be the last file in the readFilesIn listed - 1  barcode sequence is a part of mate 1 - 2  barcode sequence is a part of mate 2</td>
<td style="text-align: left;"><code>integer</code>, example: <code>0</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloCBposition</code></td>
<td style="text-align: left;">position of Cell Barcode(s) on the barcode read. Presently only works with soloType CB_UMI_Complex, and barcodes are assumed to be on Read2. Format for each barcode: startAnchor_startPosition_endAnchor_endPosition start(end)Anchor defines the Anchor Base for the CB: 0: read start; 1: read end; 2: adapter start; 3: adapter end start(end)Position is the 0-based position with of the CB start(end) with respect to the Anchor Base String for different barcodes are separated by space. Example: inDrop (Zilionis et al, Nat. Protocols, 2017): soloCBposition 0_0_2_-1 3_1_3_8</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloUMIposition</code></td>
<td style="text-align: left;">position of the UMI on the barcode read, same as soloCBposition Example: inDrop (Zilionis et al, Nat. Protocols, 2017): soloCBposition 3_9_3_14</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloAdapterSequence</code></td>
<td style="text-align: left;">adapter sequence to anchor barcodes. Only one adapter sequence is allowed.</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloAdapterMismatchesNmax</code></td>
<td style="text-align: left;">maximum number of mismatches allowed in adapter sequence.</td>
<td style="text-align: left;"><code>integer</code>, example: <code>1</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloCBmatchWLtype</code></td>
<td style="text-align: left;">matching the Cell Barcodes to the WhiteList - Exact  only exact matches allowed - 1MM  only one match in whitelist with 1 mismatched base allowed. Allowed CBs have to have at least one read with exact match. - 1MM_multi  multiple matches in whitelist with 1 mismatched base allowed, posterior probability calculation is used choose one of the matches. Allowed CBs have to have at least one read with exact match. This option matches best with CellRanger 2.2.0 - 1MM_multi_pseudocounts  same as 1MM_Multi, but pseudocounts of 1 are added to all whitelist barcodes. - 1MM_multi_Nbase_pseudocounts  same as 1MM_multi_pseudocounts, multimatching to WL is allowed for CBs with N-bases. This option matches best with CellRanger &gt;= 3.0.0 - EditDist_2  allow up to edit distance of 3 fpr each of the barcodes. May include one deletion + one insertion. Only works with soloType CB_UMI_Complex. Matches to multiple passlist barcdoes are not allowed. Similar to ParseBio Split-seq pipeline.</td>
<td style="text-align: left;"><code>string</code>, example: <code>"1MM_multi"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloInputSAMattrBarcodeSeq</code></td>
<td style="text-align: left;">when inputting reads from a SAM file (readsFileType SAM SE/PE), these SAM attributes mark the barcode sequence (in proper order). For instance, for 10X CellRanger or STARsolo BAMs, use soloInputSAMattrBarcodeSeq CR UR . This parameter is required when running STARsolo with input from SAM.</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloInputSAMattrBarcodeQual</code></td>
<td style="text-align: left;">when inputting reads from a SAM file (readsFileType SAM SE/PE), these SAM attributes mark the barcode qualities (in proper order). For instance, for 10X CellRanger or STARsolo BAMs, use soloInputSAMattrBarcodeQual CY UY . If this parameter is - (default), the quality H will be assigned to all bases.</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloStrand</code></td>
<td style="text-align: left;">strandedness of the solo libraries: - Unstranded  no strand information - Forward  read strand same as the original RNA molecule - Reverse  read strand opposite to the original RNA molecule</td>
<td style="text-align: left;"><code>string</code>, example: <code>"Forward"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloFeatures</code></td>
<td style="text-align: left;">genomic features for which the UMI counts per Cell Barcode are collected - Gene  genes: reads match the gene transcript - SJ  splice junctions: reported in SJ.out.tab - GeneFull  full gene (pre-mRNA): count all reads overlapping genes exons and introns - GeneFull_ExonOverIntron  full gene (pre-mRNA): count all reads overlapping genes exons and introns: prioritize 100% overlap with exons - GeneFull_Ex50pAS  full gene (pre-RNA): count all reads overlapping genes exons and introns: prioritize &gt;50% overlap with exons. Do not count reads with 100% exonic overlap in the antisense direction.</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"Gene"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloMultiMappers</code></td>
<td style="text-align: left;">counting method for reads mapping to multiple genes - Unique  count only reads that map to unique genes - Uniform  uniformly distribute multi-genic UMIs to all genes - Rescue  distribute UMIs proportionally to unique+uniform counts (~ first iteration of EM) - PropUnique  distribute UMIs proportionally to unique mappers, if present, and uniformly if not. - EM  multi-gene UMIs are distributed using Expectation Maximization algorithm</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"Unique"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloUMIdedup</code></td>
<td style="text-align: left;">type of UMI deduplication (collapsing) algorithm - 1MM_All  all UMIs with 1 mismatch distance to each other are collapsed (i.e.&nbsp;counted once). - 1MM_Directional_UMItools  follows the directional method from the UMI-tools by Smith, Heger and Sudbery (Genome Research 2017). - 1MM_Directional  same as 1MM_Directional_UMItools, but with more stringent criteria for duplicate UMIs - Exact  only exactly matching UMIs are collapsed. - NoDedup  no deduplication of UMIs, count all reads. - 1MM_CR  CellRanger2-4 algorithm for 1MM UMI collapsing.</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"1MM_All"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloUMIfiltering</code></td>
<td style="text-align: left;">type of UMI filtering (for reads uniquely mapping to genes) - -  basic filtering: remove UMIs with N and homopolymers (similar to CellRanger 2.2.0). - MultiGeneUMI  basic + remove lower-count UMIs that map to more than one gene. - MultiGeneUMI_All  basic + remove all UMIs that map to more than one gene. - MultiGeneUMI_CR  basic + remove lower-count UMIs that map to more than one gene, matching CellRanger &gt; 3.0.0 . Only works with soloUMIdedup 1MM_CR</td>
<td style="text-align: left;">List of <code>string</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloOutFileNames</code></td>
<td style="text-align: left;">file names for STARsolo output: file_name_prefix gene_names barcode_sequences cell_feature_count_matrix</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"Solo.out/", "features.tsv", "barcodes.tsv", "matrix.mtx"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloCellFilter</code></td>
<td style="text-align: left;">cell filtering type and parameters - None  do not output filtered cells - TopCells  only report top cells by UMI count, followed by the exact number of cells - CellRanger2.2  simple filtering of CellRanger 2.2. Can be followed by numbers: number of expected cells, robust maximum percentile for UMI count, maximum to minimum ratio for UMI count The harcoded values are from CellRanger: nExpectedCells=3000; maxPercentile=0.99; maxMinRatio=10 - EmptyDrops_CR  EmptyDrops filtering in CellRanger flavor. Please cite the original EmptyDrops paper: A.T.L Lun et al, Genome Biology, 20, 63 (2019): https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y Can be followed by 10 numeric parameters: nExpectedCells maxPercentile maxMinRatio indMin indMax umiMin umiMinFracMedian candMaxN FDR simN The harcoded values are from CellRanger: 3000 0.99 10 45000 90000 500 0.01 20000 0.01 10000</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"CellRanger2.2", "3000", "0.99", "10"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>--soloOutFormatFeaturesGeneField3</code></td>
<td style="text-align: left;">field 3 in the Gene features.tsv file. If -, then no 3rd field is output.</td>
<td style="text-align: left;">List of <code>string</code>, example: <code>"Gene Expression"</code>, multiple_sep: <code>";"</code></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>--soloCellReadStats</code></td>
<td style="text-align: left;">Output reads statistics for each CB - Standard  standard output</td>
<td style="text-align: left;"><code>string</code></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="authors" class="level2">
<h2 class="anchored" data-anchor-id="authors">Authors</h2>
<ul>
<li><p>Angela Oliveira Pisco <a href="https://github.com/aopisco"><i class="fa-brands fa-github" aria-label="github"></i></a> <a href="https://orcid.org/0000-0003-0142-2355"><i class="fa-brands fa-orcid" aria-label="orcid"></i></a> <a href="https://linkedin.com/in/aopisco"><i class="fa-brands fa-linkedin" aria-label="linkedin"></i></a> (author)</p></li>
<li><p>Robrecht Cannoodt <a href="https://github.com/rcannood"><i class="fa-brands fa-github" aria-label="github"></i></a> <a href="https://orcid.org/0000-0003-3641-729X"><i class="fa-brands fa-orcid" aria-label="orcid"></i></a> <a href="https://linkedin.com/in/robrechtcannoodt"><i class="fa-brands fa-linkedin" aria-label="linkedin"></i></a> (author, maintainer)</p></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    if (id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        for (let i = 0; i < 2; i++) {
          container.appendChild(note.children[i].cloneNode(true));
        }
        return container.innerHTML
      } else {
        return note.innerHTML;
      }
    } else {
      return note.innerHTML;
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      try { hash = new URL(url).hash; } catch {}
      const id = hash.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note !== null) {
        try {
          const html = processXRef(id, note);
          instance.setContent(html);
        } finally {
          instance.enable();
          instance.show();
        }
      } else {
        // See if we can fetch this
        fetch(url.split('#')[0])
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.getElementById(id);
          console.log(htmlDoc.body.innerHTML);
          if (note !== null) {
            const html = processXRef(id, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p> OpenPipelines.bio 2022</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/openpipelines-bio/website/edit/main/components/modules/mapping/star_align_v273a.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/openpipelines-bio/website/issues/new" class="toc-action">Report an issue</a></p></div></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>