---
title: "Freemuxlet"
namespace: "Genetic demux"
description: "Freemuxlet is a software tool to deconvolute sample identity and identify multiplets when multiple samples are pooled by barcoded single cell sequencing."
type: "module"
toc: false
---

::: {.column-margin}

### Info

ID: `freemuxlet`  
Namespace: `genetic_demux`

### Links
[Source](https://github.com/openpipelines-bio/openpipeline/blob/1.0.0-rc5/./src/genetic_demux/freemuxlet){.btn-action .btn-sm .btn-info .btn role="button"}

:::

If external genotyping
data is not available, the genotyping-free version demuxlet, freemuxlet, would be recommended

## Example commands

You can run the pipeline using `nextflow run`.

### View help

You can use `--help` as a parameter to get an overview of the possible parameters.

```bash
nextflow run openpipelines-bio/openpipeline \
  -r 1.0.0-rc5 -latest \
  -main-script target/nextflow/genetic_demux/freemuxlet/main.nf \
  --help
```

### Run command

<details>
<summary>Example of `params.yaml`</summary>
```yaml
# Input
# plp: "foo"
# init_cluster: "path/to/file"
nsample: 2
aux_files: false
verbose: 100
doublet_prior: 0.5
geno_error: 0.1
bf_thres: 5.41
frac_init_clust: 1
iter_init: 10
keep_init_missing: false
randomize_singlet_score: false
seed: 0
cap_bq: 20
min_bq: 13
# group_list: "foo"
min_total: 0
min_umi: 0
min_snp: 0

# Output
# output: "$id.$key.output.output"
# out: "freemuxlet"

# Nextflow input-output arguments
publish_dir: # please fill in - example: "output/"
# param_list: "my_params.yaml"
```
</details>

```bash
nextflow run openpipelines-bio/openpipeline \
  -r 1.0.0-rc5 -latest \
  -profile docker \
  -main-script target/nextflow/genetic_demux/freemuxlet/main.nf \
  -params-file params.yaml
```

:::{.callout-note}
Replace `-profile docker` with `-profile podman` or `-profile singularity` depending on the desired backend.
:::

## Argument groups

### Input

|Name                        |Description                                                                                                                          |Attributes                |
|:---------------------------|:------------------------------------------------------------------------------------------------------------------------------------|:-------------------------|
|`--plp`                     |Prefix of input files generated by dsc-pileup                                                                                        |`string`                  |
|`--init_cluster`            |Input file containing the initial cluster information.                                                                               |`file`                    |
|`--nsample`                 |Number of samples multiplexed together                                                                                               |`integer`, default: `2`   |
|`--aux_files`               |Turn on writing auxilary output files                                                                                                |`boolean_true`            |
|`--verbose`                 |Turn on verbose mode with specific verbosity threshold. 0: fully verbose, 100 : no verbose messages.                                 |`integer`, default: `100` |
|`--doublet_prior`           |Prior of doublet.                                                                                                                    |`double`, default: `0.5`  |
|`--geno_error`              |Genotype error parameter per cluster.                                                                                                |`double`, default: `0.1`  |
|`--bf_thres`                |Bayes Factor Threshold used in the initial clustering.                                                                               |`double`, default: `5.41` |
|`--frac_init_clust`         |Fraction of droplets to be clustered in the very first round of initial clustering procedure.                                        |`double`, default: `1`    |
|`--iter_init`               |Iteration for initial cluster assignment (set to zero to skip the iterations).                                                       |`integer`, default: `10`  |
|`--keep_init_missing`       |Keep missing cluster assignment as missing in the initial iteration.                                                                 |`boolean_true`            |
|`--randomize_singlet_score` |Randomize the singlet scores to test its effect.                                                                                     |`boolean_true`            |
|`--seed`                    |Seed for random number (use clocks if not set).                                                                                      |`integer`, default: `0`   |
|`--cap_bq`                  |Maximum base quality (higher BQ will be capped).                                                                                     |`integer`, default: `20`  |
|`--min_bq`                  |Minimum base quality to consider (lower BQ will be skipped).                                                                         |`integer`, default: `13`  |
|`--group_list`              |List of tag readgroup/cell barcode to consider in this run. All other barcodes will be ignored. This is useful for parallelized run. |`string`                  |
|`--min_total`               |Minimum number of total reads for a droplet/cell to be considered.                                                                   |`integer`, default: `0`   |
|`--min_umi`                 |Minimum number of UMIs for a droplet/cell to be considered.                                                                          |`integer`, default: `0`   |
|`--min_snp`                 |Minimum number of SNPs with coverage for a droplet/cell to be considered.                                                            |`integer`, default: `0`   |


### Output

|Name       |Description                   |Attributes                        |
|:----------|:-----------------------------|:---------------------------------|
|`--output` |Output directory              |`file`, example: `"freemux"`      |
|`--out`    |freemuxlet Output file prefix |`string`, example: `"freemuxlet"` |

## Authors


  * Xichen Wu [{{< fa brands github >}}](https://github.com/wxicu) [{{< fa brands orcid >}}](https://orcid.org/0009-0008-2168-4508) [{{< fa brands linkedin >}}](https://linkedin.com/in/xichen-wu) (author)
